ballerina/dispatcher_service.bal

```diff
@@ -0,0 +1,324 @@
+// Copyright (c) 2025 WSO2 LLC (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/uuid;
+
+# Represents the dispatcher service type definition.
+type DispatcherService distinct service object {
+    *http:Service;
+
+    isolated function addServiceRef(Service|AdvancedService mcpService);
+    isolated function removeServiceRef();
+};
+
+DispatcherService dispatcherService = isolated service object {
+    private Service|AdvancedService? mcpService = ();
+    private boolean isInitialized = false;
+    private string? sessionId = ();
+
+    isolated function addServiceRef(Service|AdvancedService mcpService) {
+        lock {
+            self.mcpService = mcpService;
+        }
+    }
+
+    isolated function removeServiceRef() {
+        lock {
+            self.mcpService = ();
+        }
+    }
```

Comment:
Are we saying there can only be one MCP service always? Why do we need that restriction?